src/components/planttracker/PlantTracker.java

```diff
@@ -51,29 +59,39 @@ public List<String> listAllPlants() {
         return plantNames;
     }
 
-    @Override
+    /**
+     * Water the plant.
+     *
+     * @param name The name of the plant to water.
+     */
     public void waterPlant(String name) {
         Plant plant = this.plants.get(name);
         if (plant != null) {
             logger.info("Watering " + name + "...");
+            plant.water();
             logger.info(name + " has been watered.");
         } else {
             logger.warning("Plant not found: " + name);
         }
     }
 
-    @Override
+    /**
+     * Show care instructions for the specified plant.
+     *
+     * @param name The name of the plant.
+     */
     public void showCareInstructions(String name) {
         Plant plant = this.plants.get(name);
         if (plant != null) {
-            logger.info("Care instructions for " + name + ": "
-                    + plant.getCareInstructions());
+            logger.info("Care instructions for " + name + ": " + plant.getCareInstructions());
         } else {
             logger.warning("Plant not found: " + name);
         }
     }
 
-    @Override
+    /**
+     * Display all plants with detailed information.
+     */
     public void displayAllPlants() {
```

Comment:
From a layering standpoint, I think this is probably a secondary method. Also, the documentation isn't needed if you include that `@Override` tag.